modules/libraries/src/backend/utils.ts

```diff
@@ -0,0 +1,178 @@
+        PKG_EXECPATH: 'PKG_INVOKE_NODEJS'
+      }
+    })
+
+    const resultBuffer: string[] = []
+
+    spawned.stdout.on('data', msg => resultBuffer.push(msg.toString()))
+    spawned.stderr.on('data', msg => resultBuffer.push(msg.toString()))
+
+    await Promise.fromCallback(cb => spawned.stdout.on('close', cb))
+
+    return resultBuffer.join('')
+  } catch (err) {
+    console.error('error ', err)
+  }
+}
+
+export const createNodeSymlink = async () => {
+  const nodePath = path.join(path.dirname(process.execPath), 'node')
+
+  // The symlink is only necessary when running the binary and node is not installed
+  if (process.execPath.endsWith('bp.exe') || process.execPath.endsWith('bp')) {
+    if (!(await fse.pathExists(nodePath))) {
+      await fse.symlink(process.execPath, nodePath)
+    }
+  }
+}
+
+export const syncAllFiles = async (bp: typeof sdk) => {
+  const files = await bp.ghost.forGlobal().directoryListing(LIB_FOLDER, '*.*')
+  await Promise.mapSeries(files, file => copyFileLocally(file, bp))
+}
+
+export const copyFileLocally = async (fileName: string, bp: typeof sdk): Promise<boolean> => {
+  if (!(await bp.ghost.forGlobal().fileExists(LIB_FOLDER, fileName))) {
+    return false
+  }
+
+  try {
+    const fileContent = await bp.ghost.forGlobal().readFileAsBuffer(LIB_FOLDER, fileName)
+    await fse.writeFile(path.join(sharedLibsDir, fileName), fileContent)
+    return true
+  } catch (err) {
+    bp.logger.error(`Couldn't copy locally. ${err}`)
+    return false
+  }
+}
+
+const deleteLibraryArchive = async (filename: string, bp: typeof sdk) => {
+  try {
+    if (await bp.ghost.forGlobal().fileExists(LIB_FOLDER, filename)) {
+      await bp.ghost.forGlobal().deleteFile(LIB_FOLDER, filename)
+    }
+
+    if (await fse.pathExists(path.join(sharedLibsDir, filename))) {
+      await fse.remove(path.join(sharedLibsDir, filename))
+    }
+  } catch (err) {
+    bp.logger.warn(`Error while deleting the library archive ${err}`)
+  }
+}
+
+export const removeLibrary = async (name: string, bp: typeof sdk): Promise<boolean> => {
+  const packageContent = JSON.parse(await fse.readFile(packageJsonPath, 'UTF-8'))
+  const source = packageContent.dependencies[name]
+
+  if (!source) {
+    return false
+  }
+
+  if (source.endsWith('.tgz')) {
+    await deleteLibraryArchive(source.replace('file:', ''), bp)
+  }
+
+  delete packageContent.dependencies[name]
+  await fse.writeFile(packageJsonPath, JSON.stringify(packageContent, undefined, 2))
+
+  return true
+}
+
+export const publishPackageChanges = async (bp: typeof sdk) => {
+  const packageContent = await fse.readFile(packageJsonPath, 'UTF-8')
+  await bp.ghost.forGlobal().upsertFile(LIB_FOLDER, 'package.json', packageContent)
+
+  const packageLockContent = await fse.readFile(packageLockJsonPath, 'UTF-8')
+  await bp.ghost.forGlobal().upsertFile(LIB_FOLDER, 'package-lock.json', packageLockContent)
+}
+
+export const createDefaultExample = async (bp: typeof sdk) => {
```

Comment:
I get that, but it's a bit weird  to hold the example in the code itself. I know this suggestion doesn't change anything but can you write it in an independent .txt file ?